hubble/src/indexer/mod.rs

```diff
@@ -56,25 +56,34 @@ where
 
             match self
                 .create_fetcher_client(&mut join_set, self.context.clone())
+                .instrument(info_span!("creator"))
                 .await
             {
                 Some(fetcher_client) => {
                     let self_clone = self.clone();
                     let fetcher_client_clone = fetcher_client.clone();
-                    join_set
-                        .spawn(async move { self_clone.run_fetcher(fetcher_client_clone).await });
+                    join_set.spawn(
+                        async move { self_clone.run_fetcher(fetcher_client_clone).await }
+                            .instrument(info_span!("fetcher")),
+                    );
 
                     let self_clone = self.clone();
                     let fetcher_client_clone = fetcher_client.clone();
-                    join_set
-                        .spawn(async move { self_clone.run_finalizer(fetcher_client_clone).await });
+                    join_set.spawn(
+                        async move { self_clone.run_finalizer(fetcher_client_clone).await }
+                            .instrument(info_span!("finalizer")),
+                    );
 
                     let self_clone = self.clone();
                     let fetcher_client_clone = fetcher_client.clone();
-                    join_set.spawn(async move { self_clone.run_fixer(fetcher_client_clone).await });
+                    join_set.spawn(
+                        async move { self_clone.run_fixer(fetcher_client_clone).await }
+                            .instrument(info_span!("fixer")),
+                    );
 
                     if let EndOfRunResult::Exit = self
                         .handle_end_of_run(&mut join_set, fetcher_client)
+                        .instrument(info_span!("exiter"))
```

Comment:
probably good to include some context in these spans as well? something like `info_span!(context = %self.context, "fetcher")`

Reply:
this behaviour is not guaranteed, and may change depending on how the tasks are spread across threads. i believe you can use [`in_current_span`](https://docs.rs/tracing/latest/tracing/trait.Instrument.html#method.in_current_span) to guarantee this though

Reply:
ah interesting, looks solid then 👍

Reply:
i think the part i was thinking about was:

> However, if "my_future" is disabled, the spawned task will *not* be in any span.

where if you want the future to *always* be in the current span, you need to use `in_current_span`